src/services/users-to-groups.ts

```diff
@@ -71,6 +71,25 @@ export async function deleteUsersToGroups(
   userId: string,
   usersToGroupsId: string,
 ) {
+  const groupToLeave = await dbPool.query.usersToGroups.findFirst({
+    where: eq(db.usersToGroups.id, usersToGroupsId),
+  });
+
+  if (!groupToLeave) {
+    throw new Error('Users to Groups not found');
+  }
+
+  const isGroupLeader = await dbPool.query.registrations.findFirst({
+    where: and(
+      eq(db.registrations.userId, userId),
+      eq(db.registrations.groupId, groupToLeave.groupId),
+    ),
+  });
+
+  if (isGroupLeader) {
+    throw new Error('Leader is not allowed to leave the Group');
+  }
```

Comment:
maybe we change this logic to explain that a registration is attached to this group.

rename isGroupLeader to isRegistrationAttached

  if (isGroupLeader) {
    throw new Error('Can not leave group with a registration attached to it');
  }